torch/nn/functional.py

```diff
@@ -3965,7 +3981,11 @@ def multi_head_attention_forward(query,                           # type: Tensor
     assert list(attn_output_weights.size()) == [bsz * num_heads, tgt_len, src_len]
 
     if attn_mask is not None:
-        attn_output_weights += attn_mask
+        if attn_mask.dtype == torch.bool:
+            attn_output_weights.masked_fill_(attn_mask, float('-inf'))
```

Comment:
Are you sure this doesn't work with ```uint8```? If it does, we can avoid the type conversion. Or, since both BoolTensor and ByteTensor are backed by uint8 the conversion might even be free and this doesn't matter.

Reply:
I think, uint8<->bool conversion currently still creates a clone

Reply:
@zhangguanheng66 - It'd be good to investigate this. Does uint8 <-> bool incur a memory copy / clone or is it just a metadata change?

Reply:
a memory copy.
`masked_fill_` does support byte tensor. However, it is depreciated with a warning. And this is the main reason to convert a byte tensor to bool tensor.

Reply:
If it is deprecated with a warning we should call it directly or re-raise the warning ourselves. If masked_fill_ isn't support ByteTensor anymore, neither should we. Warnings can be turned off or the conversion can happen at the callee's level. So let's not do the conversion if this is the case.

Reply:
Because then eventually we'll need to go through a deprecation phase as well.